### ISSUE TITLE
No IntelliSense & parameter hints for calls to local functions

### ISSUE THREAD

**Issue by cateyes99** [OmniSharp/omnisharp-vscode#1715]
## Environment data
`dotnet --info` output:
```
.NET Command Line Tools (1.0.1)

Product Information:
 Version:            1.0.1
 Commit SHA-1 hash:  005db40cd1

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.11
 OS Platform: Darwin
 RID:         osx.10.11-x64
 Base Path:   /usr/local/share/dotnet/sdk/1.0.1
```
VS Code version: `1.14.2`
C# Extension version: `1.11.0`

## Steps to reproduce
Create a local function, and make a call to it:
```C#
static void main(string[] args)
{
    var flag = LocalFunction(1);
    bool LocalFunction(int i)
    {
        return i > 0;
    }
}
```
## Expected  behavior
1. When typing arguments for the call to the **local function**, in `var flag = LocalFunction(1);`, **IntelliSense** should be provided for the **parameter hints**.

1. When using the feature `Trigger Parameter Hints` <kbd>⇧</kbd><kbd>⌘</kbd><kbd>Space</kbd> it should trigger the **parameter hints** for the call to the **local function**.

## Actual behavior
No IntelliSense provided for **1**, and no parameter hints triggered for **2**.

**Comment by rchande** [comment]
This is another good one for @akshita31 to look at (cc @DustinCampbell @TheRealPiotrP )

**Comment by DustinCampbell** [comment]
Agreed. @akshita31, I went ahead and assigned this to you.

**Comment by akshita31** [comment]
The problem here is that in the SignatureHelpService, we are first finding the containingType for the given invocation and then returning all the methods in that containingType with the same name as the method. In case of localFunction , the containing type is being returned as the Class name and not the main function .But I am not sure whether this is the problem with roslyn returning the incorrect containingType or a problem with the way we are finding the methodOverloads in omnisharp-roslyn.

**Comment by DustinCampbell** [comment]
The fix for this issue has been merged into OmniSharp. Once we take a new build of OmniSharp, it will appear here.

**Comment by akshita31** [comment]
This issue has been fixed in the 1.14 milestone. We just released a beta of 1.14 - https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.14.0-beta4. Follow [these instructions](https://github.com/OmniSharp/omnisharp-vscode/releases/tag/v1.14.0-beta4) to a beta release of C# for VS Code.
